QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,190 @@
+
+`rmw_implementation` does not currently track test coverage.
+
+### Performance [4.iv]
+
+`rmw_implementation` does not currently have performance tests.
+
+### Linters and Static Analysis [4.v]
+
+`rmw_implementation` uses and passes all the standard linters and static analysis tools for a C++ package as described in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#linters-and-static-analysis).
+
+Currently nightly results can be seen here:
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/rmw_implementation/)
+* [linux-arm64_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rmw_implementation/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rmw_implementation/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rmw_implementation/)
+
+## Dependencies [5]
+
+### Direct Runtime ROS Dependencies [5.i/5.ii]
+
+`rmw_implementation` has run-time and build-time dependencies that need to be considered for this declaration.
+
+ - rcpputils
+ - rcutils
+ - rmw
+ - rmw_implementation_cmake
+
+### Direct Runtime non-ROS Dependency [5.iii]
+
+`rosidl_typesupport_cpp` does not have any runtime non-ROS dependencies.
+
+It has one "buildtool" dependency, which do not affect the resulting quality of the package, because they do not contribute to the public library API.
+It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.
+
+## Platform Support [6]
+
+`rmw_implementation` supports all of the tier 1 platforms as described in [REP-2000](https://www.ros.org/reps/rep-2000.html#support-tiers), and tests each change against all of them.
+
+Currently nightly results can be seen here:
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/rmw_implementation/)
+* [linux-arm64_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rmw_implementation/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rmw_implementation/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rmw_implementation/)
+
+## Vulnerability Disclosure Policy [7.i]
+
+This package does not yet have a Vulnerability Disclosure Policy
+
+# Current status Summary
+
+The chart below compares the requirements in the REP-2004 with the current state of the rcutils package.
+
+|Number|  Requirement| Current state |
+|--|--|--|
+|1| **Version policy** |---|
+|1.i|Version Policy available | ✓ |
+|1.ii|Stable version |x|
+|1.iii|Declared public API|None|
+|1.iv|API stability policy|None|
+|1.v|ABI stability policy|None|
+|1.vi_|API/ABI stable within ros distribution|None|
```

Comment:
The policies are there, although these don't apply directly to the package. I think these can be marked with a check.
```suggestion
|1.iii|Declared public API|Not required|
|1.iv|API stability policy|✓|
|1.v|ABI stability policy|✓|
|1.vi_|API/ABI stable within ros distribution|✓|
```